pubsub.go

```diff
@@ -126,8 +170,14 @@ func (p *PubsubValueStore) Subscribe(key string) error {
 	// record hasn't expired.
 	//
 	// Also, make sure to do this *before* subscribing.
-	p.ps.RegisterTopicValidator(topic, func(ctx context.Context, _ peer.ID, msg *pubsub.Message) bool {
-		return p.isBetter(key, msg.GetData())
+	myID := p.host.ID()
+	_ = p.ps.RegisterTopicValidator(topic, func(ctx context.Context, src peer.ID, msg *pubsub.Message) bool {
+		isBetter, isEqual := p.isBetter(key, msg.GetData())
+
+		if src == myID && isEqual {
```

Comment:
If a node tries to send a repeat message let them do it, it's an active decision on their part. Otherwise, don't propagate repeats.